src/block.rs

```diff
@@ -170,46 +170,185 @@ impl BlockBuilder {
 
 #[cfg(test)]
 mod tests {
+    use rstest::rstest;
+
     use super::*;
-    use crate::test_utils::{gen_attrs, gen_empty_attrs};
+    use crate::{
+        test_utils::{assert_debug_snapshot, decode_codec_entries},
+        types::ValueDeletable,
+    };
 
-    #[test]
-    fn test_block() {
-        let mut builder = BlockBuilder::new(4096);
-        assert!(builder.add_value(b"key1", b"value1", gen_empty_attrs()));
-        assert!(builder.add_value(b"key1", b"value1", gen_empty_attrs()));
-        assert!(builder.add_value(b"key2", b"value2", gen_empty_attrs()));
-        let block = builder.build().unwrap();
-        let encoded = block.encode();
-        let decoded = Block::decode(encoded);
-        assert_eq!(block.data, decoded.data);
-        assert_eq!(block.offsets, decoded.offsets);
+    #[derive(Debug)]
+    struct BlockTestCase {
+        name: &'static str,
+        entries: Vec<RowEntry>, // Use RowEntry instead of (key, value)
+        expected_size: usize,   // Expected size of the block
     }
 
-    #[test]
-    fn test_block_with_tombstone() {
+    fn build_block(test_case: &BlockTestCase) -> Block {
         let mut builder = BlockBuilder::new(4096);
-        assert!(builder.add_value(b"key1", b"value1", gen_empty_attrs()));
-        assert!(builder.add_tombstone(b"key2", gen_empty_attrs()));
-        assert!(builder.add_value(b"key3", b"value3", gen_empty_attrs()));
-        let block = builder.build().unwrap();
-        let encoded = block.encode();
-        let _decoded = Block::decode(encoded);
+
+        for entry in &test_case.entries {
+            assert!(builder.add(entry.clone()));
+        }
+
+        builder.build().expect("Failed to build block")
     }
 
-    #[test]
-    fn test_block_size() {
-        let mut builder = BlockBuilder::new(4096);
-        assert!(builder.add_value(b"key1", b"value1", gen_attrs(1)));
-        assert!(builder.add_value(b"key2", b"value2", gen_attrs(1)));
-        let block = builder.build().unwrap();
-        assert_eq!(73, block.size());
+    fn assert_decoded_entries(
+        data: &Bytes,
+        offsets: &[u16],
+        expected_entries: &[RowEntry],
+    ) -> Result<(), SlateDBError> {
+        let decoded_entries = decode_codec_entries(data.clone(), offsets)?;
+        let mut prev_key = Bytes::new();
 
-        let mut builder = BlockBuilder::new(4096);
-        assert!(builder.add_value(b"key1", b"value1", gen_empty_attrs()));
-        assert!(builder.add_value(b"key2", b"value2", gen_empty_attrs()));
-        let block = builder.build().unwrap();
-        assert_eq!(57, block.size());
+        for (decoded_entry, expected_entry) in decoded_entries.iter().zip(expected_entries.iter()) {
+            // Reconstruct the full key from the key_prefix_len and key_suffix
```

Comment:
This seems rather brittle. RowEntry derives `PartialEq`. Can we use that instead?

Reply:
Good point, we could use PartialEq. After initially reviewing the code, it seems we need to implement it in **row_codec.rs**. I'd like to review the changes with you, if my thinking is right as shown below, to ensure it aligns with your expectations. 

```rust
impl PartialEq<RowEntry> for SstRowEntry {
    fn eq(&self, other: &RowEntry) -> bool {
        let full_key = if self.key_prefix_len > 0 {
            let mut full_key = other.key.slice(..self.key_prefix_len).to_vec();
            full_key.extend_from_slice(&self.key_suffix);
            Bytes::from(full_key)
        } else {
            self.key_suffix.clone()
        };

        full_key == other.key
            && self.value == other.value
            && self.seq == other.seq
            && self.expire_ts == other.expire_ts
            && self.create_ts == other.create_ts
    }
}
```
Inside the **block.rs** test it will be minimized to 
```rust
let decoded_entries = decode_codec_entries(block_data.clone(), &block_offsets)
            .expect("Failed to decode codec entries");

assert_eq!(decoded_entries, test_case.entries);
```

Reply:
Is it possible to do `#[derive(PartialEq)]` rather than implement our own `eq`?

Reply:
I tried annotating the SstRowEntry with PartialEq only and ran into an issue:
```bash
assert_eq!(decoded_entries, test_case.entries);
    |  no implementation for `row_codec::SstRowEntry == types::RowEntry`
```

 Since we are using **RowEntry** in **block.rs**, and once it gets encoded, we get back an **SstRowEntry** from **row_codec.rs**. To compare both, we would need to implement **PartialEq** for SstRowEntry in a way that allows it to be compared with the **RowEntry** from block.rs and the **SstRowEntry** that we get after decoding.

Reply:
Gotcha. I'm fine with the impl you proposed then.

Reply:
Addressed.